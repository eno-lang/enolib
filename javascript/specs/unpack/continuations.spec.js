import { parse } from '../..';
import { unpack } from '../unpack.js';

const input = `
field_with_value: [value]
\\ [spaced_continuation]
| [direct_continuation]

field_with_attribute:
attribute = [value]
\\ [spaced_continuation]
| [direct_continuation]

field_with_item:
- [item]
\\ [spaced_continuation]
| [direct_continuation]

empty_field:
\\ [spaced_continuation]
| [direct_continuation]

field_with_empty_attribute:
empty_attribute =
\\ [spaced_continuation]
| [direct_continuation]

field_with_empty_item:
-
\\ [spaced_continuation]
| [direct_continuation]
`.trim();

describe('Blackbox test', () => {
    describe('Continuations', () => {
        it('performs as expected', () => {
            const document = parse(input);
            expect(unpack(document)).toMatchSnapshot();
        });
    });
});
