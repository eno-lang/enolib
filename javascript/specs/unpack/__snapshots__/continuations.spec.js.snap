// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blackbox test Continuations performs as expected 1`] = `
Object {
  "elements": Array [
    Object {
      "key": "field_with_value",
      "type": "field",
      "value": "[value] [spaced_continuation][direct_continuation]",
    },
    Object {
      "attributes": Array [
        Object {
          "key": "attribute",
          "value": "[value] [spaced_continuation][direct_continuation]",
        },
      ],
      "key": "field_with_attribute",
      "type": "field",
    },
    Object {
      "items": Array [
        Object {
          "value": "[item] [spaced_continuation][direct_continuation]",
        },
      ],
      "key": "field_with_item",
      "type": "field",
    },
    Object {
      "key": "empty_field",
      "type": "field",
      "value": "[spaced_continuation][direct_continuation]",
    },
    Object {
      "attributes": Array [
        Object {
          "key": "empty_attribute",
          "value": "[spaced_continuation][direct_continuation]",
        },
      ],
      "key": "field_with_empty_attribute",
      "type": "field",
    },
    Object {
      "items": Array [
        Object {
          "value": "[spaced_continuation][direct_continuation]",
        },
      ],
      "key": "field_with_empty_item",
      "type": "field",
    },
  ],
  "key": null,
  "type": "section",
}
`;
